wordpress-mobile/WordPress-iOS
# Comment Threads: Add undo functionality for moderation actions

Fixes #17752 

When moderating threaded comments, an `Undo` option is now shown in the confirmation notice. When selected, the Comment is re-approved.

To test:
- Go to a Reader post you can moderate comments on.
- Select a moderation option from the drop-down menu.
  - Verify the comment is removed from the list.
  - Verify `Undo` is displayed in the confirmation notice.
- Select `Undo`.
  - Verify the comment is approved.
  - Verify the comment is added to the list.
  - Verify a confirmation notice is displayed without the `Undo` option.

https://user-images.githubusercontent.com/1816888/149399650-82653196-a2c5-416e-8c96-5eda4bc3fb9d.mp4

## Regression Notes
1. Potential unintended areas of impact
Should be none. This is a new feature.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested functionality.

3. What automated tests I added (or what prevented me from doing so)
N/A. 

PR submission checklist:
